cmake/os/SunOS.cmake

```diff
@@ -17,6 +17,10 @@ INCLUDE(CheckSymbolExists)
 INCLUDE(CheckCSourceRuns)
 INCLUDE(CheckCSourceCompiles) 
 
+# fcntl(fd, F_SETFL, O_DIRECT) is not supported,
+# and directio(3C) would only work on UFS or NFS, not ZFS.
+SET(HAVE_FCNTL_DIRECT 0 CACHE INTERNAL "")
```

Comment:
I checked a few other POSIX-like operating systems. OpenBSD does not define `O_DIRECT` at all. FreeBSD and NetBSD allow `O_DIRECT` for both `open` and `fcntl`. For IBM AIX, `O_DIRECT` is documented for `open` but not for `fcntl()`.

@grooverdan Should we do something similar in `cmake/os/AIX.cmake`? And perhaps refactor the code so that we will pass the `O_DIRECT` flag already to `open`, like it was before cf1cf3a0d1fe277d18dae782b164f6c99f81be83?

Reply:
Addressed in 3dd8648073c390e843bfbabe428841fad987442b.
https://buildbot.mariadb.org/#/builders/121/builds/17538 is a successful IBM AIX run of this. In the build log there is no mention of looking for `O_DIRECT`; there is one about `O_NONBLOCK`. In the FreeBSD 14 log https://buildbot.mariadb.org/#/builders/619/builds/654 we can find attempts to look up both symbols:
```
-- Looking for O_NONBLOCK
-- Looking for O_NONBLOCK - found
-- Looking for O_DIRECT
-- Looking for O_DIRECT - found
```
This makes me believe that this approach should work on Solaris as well.